File: Codevita/round2/b.cpp
#include <bits/stdc++.h>
#include <ctype.h>
using namespace std;

typedef vector<bool> vb;
typedef vector<int> vi;
typedef pair<int, int> ii;
typedef vector<ii> vii;
typedef vector<vii> graph;
typedef long long ll;
typedef unsigned long long ull;

#define TESTCASE(T) cout << "Case #" << T << ": "
#define print(x) cout << x << endl
#define max_ele(x) max_element(x.begin(), x.end())

void selsort (vector<string>& a, vector<int>& b, int count)
{
  vector<int>::iterator j = b.begin ();
  vector<string>::iterator k = a.begin ();
  for (int i = 0; i < count; i++)
  {
    if (i & 1)  // Sort numbers
    {
      iter_swap (min_element (j, b.end ()), j);
      j++;
    }
    else    // Sort words
    {
      iter_swap (min_element (k, a.end ()), k);
      k++;
    }
  }
}
int main()
{
  int t;
  cin >> t;
  for (int T = 0; T < t; T++)
  {
    vector<string> a;
    vector<int> b;
    string inp, s, prev;
    int num, i = 0, count;
//    getline (cin, inp, '\n');
//    stringstream in(inp);
    while (cin >> s)
    {
        string::const_iterator k = s.begin();
      if (i & 1)
      {
        num = atoi (s.c_str ());
        b.push_back (num);
      }
      else
      {
        if(isdigit( *k))
        {
            count=atoi(s.c_str());
            break;
        }
        std::transform(s.begin(), s.end(), s.begin(), ::tolower);
        a.push_back (s);
      }
      i++;
    }
    selsort (a, b, count);
    int idx1 = 0, idx2 = 0, range = 2 * b.size ();
    for (int r = 0; r < range; r++)
    {
      if (r & 1)
      {
        cout << b[idx1] << " ";
        idx1++;
      }
      else
      {
        cout << a[idx2] << " ";
        idx2++;
      }
     }
     if (T != t - 1)
       cout << endl;

  }
  return 0;
}






/*#include<iostream>
using namespace std;



int main()
{
        int n;
        cin>>n;
        while(n--)
        {
            string s;
            cin>>s;
            int st;
            cin>>st;
            cout<<st;
        }
}
*/

File: Codevita/round2/b.py
def sel_sort (a, b, count):
    idx1 = 0
    idx2 = 0
    l = len (b)
    for i in range (count):
        if i & 1: #sort nums
            mn = b[idx1]
            mn_idx = idx1
            for j in range (idx1 + 1, l):
                if mn > b[j]:
                    mn = b[j]
                    mn_idx = j
            b[idx1], b[mn_idx] = b[mn_idx], b[idx1]
            idx1 += 1
        else:
            mn = a[idx2]
            mn_idx = idx2
            for j in range (idx2 + 1, l):
                if mn > a[j]:
                    mn = a[j]
                    mn_idx = j
            a[idx1], a[mn_idx] = a[mn_idx], a[idx1]
            idx2 += 1

n = int (input ())
for _ in range (n):
    s = input ()
    s = s.split (' ')
    a = []
    b = []
    for i in range (len(s)):
        if i & 1:
           b.append (int (s[i]))
        else:
           a.append (s[i].lower ())

    count = int (input ())
    idx1 = 0
    idx2 = 0
    sel_sort (a, b, count)
    res = []
    for i in range(len (s)):
        if i & 1:
            res.append (b[idx1])
            idx1 += 1
        else:
            res.append (a[idx1])
            idx2 += 1
    print (*res)
